src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -735,6 +710,31 @@ private Packaging.Core.PackageDependency CreateDependencyFromProject(dynamic pro
             }
         }
 
+        private static void ExtractMetadata(ProjectFactory projectFactory, Packaging.PackageBuilder builder)
```

Comment:
`projectFactory` is used to get a `TargetPath` only. Consider simplifying the method signature.
This would also help with first occurrence of this method call at line 207, where you needed to pass `this` for the first argument.